oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -568,7 +568,7 @@ private boolean visitTerm(String propertyName, String text, String boost, boolea
                 if (boost != null) {
                     fullTextQuery.boost(Float.parseFloat(boost));
                 }
-                BoolQueryBuilder shouldBoolQueryWrapper = boolQuery().should(fullTextQuery);
+                BoolQueryBuilder shouldBoolQueryWrapper = boolQuery().must(fullTextQuery);
```

Comment:
We could simplify this by removing a level of nesting: `fullTextQuery` can be passed as `must` clause at line 575, the rest (dynamic score queries) should go in the `should` clause.